dask_planner/src/parser.rs

```diff
@@ -0,0 +1,221 @@
+    }
+
+    /// Parse a SQL statement and produce a set of statements
+    pub fn parse_sql_with_dialect(
+        sql: &str,
+        dialect: &dyn Dialect,
+    ) -> Result<VecDeque<DaskStatement>, ParserError> {
+        let mut parser = DaskParser::new_with_dialect(sql, dialect)?;
+        let mut stmts = VecDeque::new();
+        let mut expecting_statement_delimiter = false;
+        loop {
+            // ignore empty statements (between successive statement delimiters)
+            while parser.parser.consume_token(&Token::SemiColon) {
+                expecting_statement_delimiter = false;
+            }
+
+            if parser.parser.peek_token() == Token::EOF {
+                break;
+            }
+            if expecting_statement_delimiter {
+                return parser.expected("end of statement", parser.parser.peek_token());
+            }
+
+            let statement = parser.parse_statement()?;
+            stmts.push_back(statement);
+            expecting_statement_delimiter = true;
+        }
+        Ok(stmts)
+    }
+
+    /// Report unexpected token
+    fn expected<T>(&self, expected: &str, found: Token) -> Result<T, ParserError> {
+        parser_err!(format!("Expected {}, found: {}", expected, found))
+    }
+
+    /// Parse a new expression
+    pub fn parse_statement(&mut self) -> Result<DaskStatement, ParserError> {
+        match self.parser.peek_token() {
+            Token::Word(w) => {
+                match w.keyword {
+                    Keyword::CREATE => {
+                        // move one token forward
+                        self.parser.next_token();
+                        // use custom parsing
+                        self.parse_create()
+                    }
+                    Keyword::DROP => {
+                        // move one token forward
+                        self.parser.next_token();
+                        // use custom parsing
+                        self.parse_drop()
+                    }
+                    _ => {
+                        // use the native parser
+                        Ok(DaskStatement::Statement(Box::from(
+                            self.parser.parse_statement()?,
+                        )))
+                    }
+                }
+            }
+            _ => {
+                // use the native parser
+                Ok(DaskStatement::Statement(Box::from(
+                    self.parser.parse_statement()?,
+                )))
+            }
+        }
+    }
+
+    /// Parse a SQL CREATE statement
+    pub fn parse_create(&mut self) -> Result<DaskStatement, ParserError> {
+        match self.parser.parse_identifier() {
+            Ok(ident) => match ident.value.to_lowercase().as_str() {
+                "create" => self.parse_create_model(),
+                _ => Ok(DaskStatement::Statement(Box::from(
+                    self.parser.parse_create()?,
+                ))),
+            },
+            Err(e) => Err(e),
+        }
```

Comment:
No need to match on the `Result`. We can just use `?` instead.

```suggestion
        let ident = self.parser.parse_identifier()?;
        match ident.value.to_lowercase().as_str() {
            "create" => self.parse_create_model(),
            _ => Ok(DaskStatement::Statement(Box::from(
                self.parser.parse_create()?,
            ))),
        }
```